server/src/main/java/org/apache/druid/server/coordinator/compact/NewestSegmentFirstIterator.java

```diff
@@ -429,8 +430,9 @@ private List<Interval> findInitialSearchInterval(
     final List<Interval> searchIntervals = new ArrayList<>();
 
     for (Interval lookupInterval : filteredInterval) {
-      if (Intervals.ETERNITY.equals(lookupInterval)) {
-        log.warn("Cannot compact datasource[%s] since interval is ETERNITY.", dataSourceName);
+      if (Intervals.ETERNITY.getStart().equals(lookupInterval.getStart())
+          || Intervals.ETERNITY.getEnd().equals(lookupInterval.getEnd())) {
+        log.warn("Cannot compact datasource[%s] since interval start or end coincides with ETERNITY.", dataSourceName);
```

Comment:
```suggestion
        log.warn("Cannot compact datasource[%s] since lookupInterval[%s] coincides with ETERNITY.", dataSourceName, lookupInterval);
```

Reply:
Done